astropy/io/fits/fitstime.py

```diff
@@ -0,0 +1,251 @@
+    """
+
+    # Global time reference coordinate keywords
+    TIME_KEYWORDS = {'TIMESYS' : 'scale',
+                     'MJDREF' : 'mjdref',
+                     'JDREF' : 'jdref',
+                     'DATEREF' : 'dateref',
+                     'TREFPOS' : 'pos',
+                     'TREFDIR' : 'dir',
+                     'TIMEUNIT' : 'unit',
+                     'TIMEOFFS' : 'offs',
+                     'OBSGEO-X' : 'loc_x',
+                     'OBSGEO-Y' : 'loc_y',
+                     'OBSGEO-Z' : 'loc_z',
+                     'DATE' : 'date',
+                     'DATE-OBS' : 'date_obs',
+                     'DATE-END' : 'date_end'}
+
+    # Column-specific time override keywords
+    COLUMN_TIME_KEYWORDS = {'TCTYP' : 'scale',
+                            'TRPOS' : 'pos',
+                            'TCUNI' : 'unit'}
+
+    # AstroPy FITS File specification
+    ASTROPY_SPECIFIER = ('+---AstroPy FITS File---+', 
+                         '/ Time information block')
+
+    # Set AstroPy Time global information
+    GLOBAL_TIME_INFO = {'TIMESYS' : ('UTC','Default time scale'),
+                        'JDREF' : (0.0,'Time columns are jd = jd1 + jd2'),
+                        'TREFPOS' : ('TOPOCENTER','Time reference position')}
+
+    # Compatibility of Time Scales and Reference Positions
+    TIME_SCALE_REF = {'tai': 'TOPOCENTER',
+                      'tt' : 'TOPOCENTER',
+                      'ut1' : 'TOPOCENTER',
+                      'utc' : 'TOPOCENTER',
+                      'tcg' : 'GEOCENTER',
+                      'tcb' : 'BARYCENTER',
+                      'tdb' : 'BARYCENTER'}
+
+    def __init__(self):
+        # Set defaults for global time scale, reference, etc.
+        self.global_info = {'scale' : 'UTC',
+                            'mjdref' : None,
+                            'unit' : 's'}
+        # Set default dictionary for time columns
+        self.time_columns = defaultdict(OrderedDict)
+        self.astropy_file = None
+
+    @classmethod
+    def is_time_column_keyword(cls, keyword):
+        """
+        Check if the FITS header keyword is a time column-specific keyword.
+
+        Parameters
+        ----------
+        keyword : str
+            FITS keyword.
+        """
+        if keyword[5:].isdigit() and keyword[:5] in cls.COLUMN_TIME_KEYWORDS:
+            return True
+        return False
+
+    def set_global_time(self, key, value, comment=None):
+        """
+        Set the global time reference frame attributes.
+
+        Parameters
+        ----------
+        key : str
+            FITS global time reference frame keyword.
+        value : int, str, list
+            value associated with specified keyword.
+        comment : str
+            comment associated with specified keyword.
+        """
+        if key in self.TIME_KEYWORDS:
+            self.global_info[self.TIME_KEYWORDS[key]] = value
+        else:
+            raise ValueError('Illegal global time keyword.')
```

Comment:
Minor suggestion: How about `ValueError('Illegal global time keyword: {} = {}'.format(key, value))` just to help the user find which one is the wrong keyword so that he/she can then edit the fits file to fix that.